File: ToyInterpreter 2/src/Model/Expression/VarExpression.java
package Model.Expression;

import Model.IDictionary;

public class VarExpression extends Expression {
	private String vname;
	@Override
	public int eval(IDictionary id) {
		return id.get(vname);
	}
	public String getVname() {
		return vname;
	}
	public void setVname(String vname) {
		this.vname = vname;
	}
	public VarExpression(String vname) {
		this.vname=vname;
	}
	@Override
	public String toString() {
		return vname;
	}
}

File: ToyInterpreter/ToyInterpreter/Model/MExpression/VarExpression.cs
using System;
namespace ToyInterpreter.Model.MExpression
{
	[Serializable()]
	public class VarExpression : Expression {
		private string vname;

		public override int eval(IDictionary id,IHeap hp) {
			return id.get(vname);
		}
		public string getVname() {
			return vname;
		}
		public void setVname(string vname) {
			this.vname = vname;
		}
		public VarExpression(string vname) {
			this.vname=vname;
		}

		public override String toString() {
			return vname;
		}
	}
}

